llama-index-core/llama_index/core/evaluation/retrieval/metrics.py

```diff
@@ -25,17 +31,55 @@ def compute(
         retrieved_texts: Optional[List[str]] = None,
         **kwargs: Any,
     ) -> RetrievalMetricResult:
-        """Compute metric."""
-        if retrieved_ids is None or expected_ids is None:
+        """Compute metric based on the provided inputs.
+
+        Parameters:
+            query (Optional[str]): The query string (not used in the current implementation).
+            expected_ids (Optional[List[str]]): Expected document IDs.
+            retrieved_ids (Optional[List[str]]): Retrieved document IDs.
+            expected_texts (Optional[List[str]]): Expected texts (not used in the current implementation).
+            retrieved_texts (Optional[List[str]]): Retrieved texts (not used in the current implementation).
+            use_granular_hit_rate (bool): If True, use the granular hit rate calculation.
+
+        Raises:
+            ValueError: If the necessary IDs are not provided.
+
+        Returns:
+            RetrievalMetricResult: The result with the computed hit rate score.
+        """
+        # Checking for the required arguments
+        if (
+            retrieved_ids is None
+            or expected_ids is None
+            or not retrieved_ids
+            or not expected_ids
+        ):
             raise ValueError("Retrieved ids and expected ids must be provided")
-        is_hit = any(id in expected_ids for id in retrieved_ids)
-        return RetrievalMetricResult(
-            score=1.0 if is_hit else 0.0,
-        )
+
+        # Determining which implementation to use based on `use_granular_hit_rate` kwarg
+        use_granular = kwargs.get("use_granular_hit_rate", False)
```

Comment:
instead of using `kwargs`, maybe should just create a class or instance attribute `use_granular`?